src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorFullyQualifiedCodeActionTranslator.ts

```diff
@@ -0,0 +1,48 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { IRazorCodeActionTranslator } from './IRazorCodeActionTranslator';
+
+export class RazorFullyQualifiedCodeActionTranslator implements IRazorCodeActionTranslator {
+
+    private static expectedCode = 'CS0246';
+
+    public applyEdit(
+        uri: vscode.Uri,
+        edit: vscode.TextEdit): [vscode.Uri | undefined, vscode.TextEdit | undefined] {
+        // The edit for this should just translate without additional help.
+        throw new Error('Method not implemented.');
+    }
+
+    public canHandleEdit(uri: vscode.Uri, edit: vscode.TextEdit): boolean {
+        // The edit for this should just translate without additional help.
+        return false;
+    }
+
+    public canHandleCodeAction(
+        codeAction: vscode.Command,
+        codeContext: vscode.CodeActionContext,
+        document: vscode.TextDocument): boolean {
+        const isMissingDiag = (value: vscode.Diagnostic) => {
+            return value.severity === vscode.DiagnosticSeverity.Error
+            && value.code === RazorFullyQualifiedCodeActionTranslator.expectedCode;
+        };
+
+        const diagnostic = codeContext.diagnostics.find(isMissingDiag);
+        if (diagnostic) {
+            const codeRange = diagnostic.range;
+            const codeValue = document.getText(codeRange);
+            if (codeAction.arguments) {
+                if (!codeAction.title.includes(' ')
```

Comment:
We could expand the number of codeActions available by essentially making this `CS0246CodeActionTranslator` and only caring about the diagnostic, but I vote that we try to only enable one at a time to avoid presentings Actions which will ultimately fail (for example. a using action would fail right now because the new edited code doesn't map to code in the .razor file)

Reply:
Can definitely do whatever you prefer. This is your baby after all. A few questions

1. Are we protected from C# changing their light bulbs texts?
1. Conceptually, what makes a code action "allowed", is it the type of code action? the diagnostic? the edit?
1. What would happen if we allowed a light bulb that we couldn't handle?

Reply:
1. I don't think we're protected, though I also don't think they're likely to change them very often.
2. Conceptually what I'm trying to allow code actions based on is type. That is, code-actions to Fully-Qualify a type are allowed, but code-actions to generate a new class based on the missing type are currently not. Unfortunately there's no machine-readable way to tell what type of code-action we're dealing with, so I'm reduced to looking at the diagnostics and messages to determine if we're looking at one we can handle.
3. If we allow a light-bulb that we couldn't handle the worst-case is that the edits from the backing file to razor file don't translate and we end up mangling the razor. Hypothetically those should match up pretty solidly, but I don't know the area well enough to rule it out. A less-bad but more easily achievable bad outcome is that we attempt to apply the code-action, the cs file doesn't have a mapping to the razor file (possibly because it's an entirely new line), so it ends up trying to edit the non-existent "backing file" and pops an error dialog about not being able to find it. I'm sure people unfamiliar with the innards of our system would be VERY confused by seeing references to non-existent files like `Index.cshtml.cs`, and possibly assume they had an old-school code-behind that they just couldn't see for some reason.

Reply:
1. I don't think we're protected, though I also don't think they're likely to change them very often.
2. Conceptually what I'm trying to allow code actions based on is type. That is, code-actions to Fully-Qualify a type are allowed, but code-actions to generate a new class based on the missing type are currently not. Unfortunately there's no machine-readable way to tell what type of code-action we're dealing with, so I'm reduced to looking at the diagnostics and messages to determine if we're looking at one we can handle.
3. If we allow a light-bulb that we couldn't handle the worst-case is that the edits from the backing file to razor file don't translate and we end up mangling the razor. Hypothetically those should match up pretty solidly, but I don't know the area well enough to rule it out. A less-bad but more easily achievable bad outcome is that we attempt to apply the code-action, the cs file doesn't have a mapping to the razor file (possibly because it's an entirely new line), so it ends up trying to edit the non-existent "backing file" and pops an error dialog about not being able to find it. I'm sure people unfamiliar with the innards of our system would be VERY confused by seeing references to non-existent files like `Index.cshtml.cs`, and possibly assume they had an old-school code-behind that they just couldn't see for some reason.